File: 1979. Find Greatest Common Divisor of Array.py
from typing import List
class Solution:
    def gcd(self,a,b):
        while(b!=0):
            a,b=b,a%b
        return a
    def findGCD(self, nums: List[int]) -> int:
        mn=nums[0]
        mx=nums[0]
        for num in nums:
            if mn>num:
                mn=num
            elif mx<num:
                mx=num
        
        return self.gcd(mx,mn)

File: 1979. Find Greatest Common Divisor of Array.cpp
class Solution {
public:
    int gcd(int a ,int b){
        while(b!=0){
            int temp=a%b;
            a=b;
            b=temp;
        }
        return a;
    }
    int findGCD(vector<int>& nums) {
        int min=nums[0];
        int max=nums[0];

        for(int num:nums){
            if(num<min)
                min=num;
            else if(num>max)
                max=num;
        }
        return gcd(max,min);
    }
};